docs/installation/sources/sources_qnx.rst

```diff
@@ -0,0 +1,295 @@
+QNX SDP is QNX's Software Development Platform which contains tools and files which are needed to cross-compile for QNX.
+
+`QNX SDP Installation Guide <https://www.qnx.com/developers/docs/7.0.0/#com.qnx.doc.qnxsdp.quickstart/topic/install_host.html>`_
+
+We will assume QNX SDP 7.1 is installed at ~/qnx710.
+
+.. _cmake_pip3_git_sw:
+
+CMake, pip3, git, dos2unix, automake, and vcstools
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+These packages provide the tools required to install *eProsima Fast DDS* and its dependencies from command line.
+
+.. code-block:: bash
+
+    sudo apt install cmake python3-pip git dos2unix automake
+
+Install vcstools for importing dependencies.
+
+.. code-block:: bash
+
+    sudo pip install -U vcstool
+
+.. _build_and_installation_sw:
+
+Build and Installation
+----------------------
+
+#. Open a terminal and clone *eProsima Fast DDS*:
+
+   .. code-block:: bash
+
+       git clone https://github.com/eProsima/Fast-DDS.git && cd Fast-DDS
+       WORKSPACE=$PWD
+
+#. Initialize git submodules for Asio, Fast-CDR and TinyXML2 and apply QNX patches to them:
+
+    .. note::
+        OpenSSL is already installed in the QNX SDP 7.1.
+
+   .. code-block:: bash
+
+       cd $WORKSPACE
+
+       # Initialize git submodules
+       git submodule update --init $WORKSPACE/thirdparty/asio/ $WORKSPACE/thirdparty/fastcdr $WORKSPACE/thirdparty/tinyxml2/
+
+       # Apply QNX patch to Asio.
+       cd $WORKSPACE/thirdparty/asio
+       git apply $WORKSPACE/build_qnx/qnx_patches/asio_qnx.patch
+
+       # Apply QNX patch to Fast-CDR.
+       cd $WORKSPACE/thirdparty/fastcdr
+       git apply $WORKSPACE/build_qnx/qnx_patches/fastcdr_qnx.patch
+
+       # Apply QNX patch to TinyXML2.
+       # TinyXML2's CMakeLists.txt has CRLF, so use unix2dos to convert the patch to CRLF. 
+       cd $WORKSPACE/thirdparty/tinyxml2
+       unix2dos $WORKSPACE/build_qnx/qnx_patches/tinyxml2_qnx.patch
+       git apply $WORKSPACE/build_qnx/qnx_patches/tinyxml2_qnx.patch
+
+#. Get foonathan_memory_vendor:
+
+   .. code-block:: bash
+
+       cd $WORKSPACE
+       vcs import < fastrtps.repos
+
+#. Optional: clone GoogleTest and apply QNX patch to it:
+
+    GoogleTest is required for building Fast-DDS tests.
+
+   .. code-block:: bash
+
+       cd $WORKSPACE
+       git clone https://github.com/google/googletest.git && cd googletest
+       git checkout v1.13.0
+       git apply $WORKSPACE/build_qnx/qnx_patches/googletest_qnx.patch
+
+#. Source the QNX environment script:
+
+   .. code-block:: bash
+
+       # We will assume the user's QNX SDP 7.1 is installed at ~/qnx710
+       ~/qnx710/qnxsdp-env.sh
+
+#. Build and install Fast-DDS and its dependencies:
+
+    .. note::
+
+       | To build examples, set COMPILE_EXAMPLES to ON in $WORKSPACE/build_qnx/common.mk.
+       | To build tests, set PERFORMANCE_TESTS, SYSTEM_TESTS, PROFILING_TESTS, and EPROSIMA_BUILD_TESTS to ON in $WORKSPACE/build_qnx/common.mk.
```

Comment:
I think it would be best to refer to [Fast DDS' CMake option documentation](https://fast-dds.docs.eprosima.com/en/latest/installation/configuration/cmake_options.html) to avoid listing too many variables.